onnxconverter_common/optimizer.py

```diff
@@ -723,6 +724,73 @@ def _build_onnx_model(node_list):
     return regenerated
 
 
+def _visit(name_to_node_map, n_name, result):
+    node = name_to_node_map[n_name]
+    if node.status == 'perm':
+        return
+    if node.status == 'temp':
+        raise Exception("This graph is not a DAG")
+    node.status = 'temp'
+    for m in node.successor:
+        if m.origin is not None:
+            _visit(name_to_node_map, m.name, result)
+    node.status = 'perm'
+    result.insert(0, node.idx)
+
+
+def _generate_next_name(prefix, idx, name_set):
+    name = prefix+str(idx)
+    while name in name_set:
+        idx = idx + 1
+        name = prefix + str(idx)
+    return name
+
+
+def _get_name(node, name_idx, prefix, name_set):
+    if hasattr(node, 'name') and node.name is not None:
+        name = node.name
+    elif node.origin is not None and node.origin.name:
+        name = node.origin.name
+    else:
+        name = _generate_next_name(prefix, name_idx, name_set)
+        name_idx = name_idx + 1
+    return name
+
+
+def _get_unmark_node(name_to_node_map):
+    for k, v in six.iteritems(name_to_node_map):
+        if v.status == 'unmark':
+            return k
+    return None
+
+
+def _topological_sort(node_list):
```

Comment:
The order is input_node at the first place of the node list?

Reply:
yes